### PR TITLE
fix(wasm): double-click-selecting a textblock would bleed into other textblocks

### LINKED ISSUE #13440: [WASM] `TextBlock` text selection using double-click bleeds into other controls
### Current behavior

When selecting text via double-click, the selection highlight is also selecting adjacent text controls. This works on WinUI Desktop but not WASM. The behavior also seems different across different browsers. Firefox does not have this problem, but Edge and Chrome do.

### Expected behavior

Double-clicking on a TextBlock should only select that text.

### How to reproduce it (as minimally and precisely as possible)

Drop this on a page and double-click on one of the TextBlocks:

    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto"/>
            <RowDefinition Height="Auto"/>
        </Grid.RowDefinitions>
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="200"/>
            <ColumnDefinition Width="200"/>
        </Grid.ColumnDefinitions>
        <TextBlock IsTextSelectionEnabled="True"
                   Text="Word"
                   Grid.Row="0"
                   Grid.Column="0"/>
        <TextBlock IsTextSelectionEnabled="True"
                   Text="Word"
                   Grid.Row="0"
                   Grid.Column="1"/>
        <TextBlock IsTextSelectionEnabled="True"
                   Text="Word"
                   Grid.Row="1"
                   Grid.Column="0"/>
        <TextBlock IsTextSelectionEnabled="True"
                   Text="Word"
                   Grid.Row="1"
                   Grid.Column="1"/>
    </Grid>

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

image

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

17.6.5

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #13440, closes https://github.com/unoplatform/Uno.Themes/issues/1459

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/WasmCSS/Uno.UI.css [hunk 1]
```diff
@@ -135,6 +135,7 @@ embed.uno-frameworkelement.uno-unarranged {
   background-clip: text !important;
   /* overflow:hidden is required for text-overflow: ellipsis to work correctly */
   overflow: hidden;
+  display: inline-flex; /* This is a workaround to prevent selection bleeding, see https://github.com/unoplatform/uno/issues/13440 */
```
Comment (MartinZikmund) [inline src/Uno.UI/WasmCSS/Uno.UI.css:138]: Does this have some potential disadvantage for us?

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI/WasmCSS/Uno.UI.css (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/WasmCSS/Uno.UI.css:138]: We **do** nest children inside `p`. A TextBlock can have child `Run`s.

**Review by ajpinedam — APPROVED** [review]
This was a good fix.

**Comment by MartinZikmund** [comment]
@ramezgerges seems some WASM test is failing here

**Comment by ramezgerges** [comment]
Switching to `inline-table` fixed the alignment problem but broke text trimming. I tried all `display` values and they either don't fix  the problem or break text trimming. This is blocked for now.

**Pull request closed without merging**